aten/src/ATen/native/cuda/jit_utils.h

```diff
@@ -77,10 +77,6 @@ template <> inline std::string typeName<c10::complex<float>>(){
 template <> inline std::string typeName<c10::complex<double>>(){
     return "std::complex<double>";
 }
-template <> inline std::string typeName<c10::complex<c10::Half>>(){
```

Comment:
Unsupported types should go to the general case which static_asserts.

https://github.com/pytorch/pytorch/pull/72576/files#diff-ac43d2ebcf5bdd28f18292c819a7503b96877b9b8a490f5da620c8fd6b8c74f4L50